sns/ut/isal_ub.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2011-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <getopt.h>
+#include <isa-l.h>
+
+#include "lib/ub.h"
+#include "lib/misc.h"
+
+#define UB_ITER 1
+#define MMAX    255
+#define KMAX    255
+
+
+static int ub_init(const char *opts M0_UNUSED)
+{
+    srand(1285360231);
+    return 0;
+}
+
+/*
+ * Generate decode matrix from encode matrix and erasure list
+ *
+ */
+
+static int gf_gen_decode_matrix_simple(uint8_t * encode_matrix,
+                    uint8_t * decode_matrix,
+                    uint8_t * invert_matrix,
+                    uint8_t * temp_matrix,
+                    uint8_t * decode_index, uint8_t * frag_err_list, int nerrs, int k,
```

Comment:
I think we need to understand the current code and then re-write these two functions.
Or, we need to include the Intel copyright information/LICENSE.

What do you think?

Reply:
I have understood the code and now I am re-writing the function. Will commit the changes once done.